domain/consensus/datastructures/consensusstatestore/utxo.go

```diff
@@ -79,12 +86,15 @@ func (css *consensusStateStore) commitVirtualUTXOSet(dbTx model.DBTransaction) e
 		return nil
 	}
 
+	css.virtualUTXOSetCache = utxolrucache.New(css.utxoSetCacheSize)
```

Comment:
Instead of creating a new cache, please add a clear() method to the cache instead.